IdentityCore/tests/MSIDBrokerOperationGetSsoCookiesResponseTests.m

```diff
@@ -0,0 +1,266 @@
+    };
+    
+    NSError *error;
+    MSIDBrokerOperationGetSsoCookiesResponse *response = [[MSIDBrokerOperationGetSsoCookiesResponse alloc] initWithJSONDictionary:json error:&error];
+    XCTAssertNil(error);
+    XCTAssertEqual(response.success, YES);
+    XCTAssertEqual(response.prtHeaders.count, 3);
+    XCTAssertEqual(response.deviceHeaders.count, 2);
+    XCTAssertTrue([ssoCookiesJsonString isEqualToString:response.jsonDictionary[@"sso_cookies"]]);
+}
+
+- (void)testInitWithJSONDictionary_whenJsonValid_EmptyDeviceHeader_shouldReturnValidResult {
+    
+    NSDictionary *ssoCookies =
+    @{
+        @"prt_headers":
+            @[
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential1": @"Base 64 Encoded JWT1"},
+                  @"home_account_id": @"uid.utid1",
+                  @"displayable_id": @"demo1@contoso.com"
+              },
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential2": @"Base 64 Encoded JWT2"},
+                  @"home_account_id": @"uid.utid2",
+                  @"displayable_id": @"demo2@contoso.com"
+              },
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential3": @"Base 64 Encoded JWT3"},
+                  @"home_account_id": @"uid.utid3",
+                  @"displayable_id":@"demo3@contoso.com"
+              }
+            ],
+         @"device_headers":@[]
+    };
+    
+    NSString *ssoCookiesJsonString = [ssoCookies msidJSONSerializeWithContext:nil];
+    
+    NSDictionary *json = @{
+        @"operation" : @"get_sso_cookies",
+        @"success" : @1,
+        MSID_BROKER_DEVICE_MODE_KEY : @"shared",
+        MSID_BROKER_WPJ_STATUS_KEY : @"joined",
+        MSID_BROKER_BROKER_VERSION_KEY : @"1.2.3",
+        @"sso_cookies" : ssoCookiesJsonString
+    };
+    
+    NSError *error;
+    MSIDBrokerOperationGetSsoCookiesResponse *response = [[MSIDBrokerOperationGetSsoCookiesResponse alloc] initWithJSONDictionary:json error:&error];
+    XCTAssertNil(error);
+    XCTAssertEqual(response.success, YES);
+    XCTAssertEqual(response.prtHeaders.count, 3);
+    XCTAssertEqual(response.deviceHeaders.count, 0);
+}
+
+- (void)testInitWithJSONDictionary_whenJsonValid_EmptyDeviceHeader_oneMissingAccounIdentifier_oneNoAccountIdentifier_shouldReturnValidResult {
+    
+    NSDictionary *ssoCookies =
+    @{
+        @"prt_headers":
+            @[
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential1": @"Base 64 Encoded JWT1"},
+                  @"home_account_id": @"uid.utid1",
+                  @"displayable_id": @"demo1@contoso.com"
+              },
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential2": @"Base 64 Encoded JWT2"},
+                  @"displayable_id": @"demo2@contoso.com"
+              },
+              @{
+                  @"header": @{@"x-ms-RefreshTokenCredential3": @"Base 64 Encoded JWT3"},
+                  @"home_account_id": @"",
```

Comment:
in MSIDPRTHeader/JsonDictionary class, didn't you return nil if home account id is an empty string?

Reply:
homeAccountId is also optional